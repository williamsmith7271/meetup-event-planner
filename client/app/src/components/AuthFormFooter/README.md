## AuthFormFooter Component
A component that shows at the bottom of the form, useful for linking to login / signup.

### Example

```js
<AuthFormFooter
  link="/login"
  text="Already a member?"
/>
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **link**    | String   |             | Any string value
| **text**    | String   |             | Any string value


### Other Information
Presentational Component
