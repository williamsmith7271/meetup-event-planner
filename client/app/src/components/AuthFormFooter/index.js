import React, { PropTypes } from 'react';
import Footer from 'grommet-udacity/components/Footer';
import Anchor from 'grommet-udacity/components/Anchor';

const AuthFormFooter = ({
  link,
  text,
}) => (
  <Footer align="center" justify="center">
    <span>{`${text}`}{'  '}</span>
    <Anchor href={link}>
      {`${link.charAt(1).toUpperCase()}${link.slice(2)}`}
    </Anchor>
  </Footer>
);

AuthFormFooter.propTypes = {
  link: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default AuthFormFooter;
