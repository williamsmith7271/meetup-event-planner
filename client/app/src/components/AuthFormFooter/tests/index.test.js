import AuthFormFooter from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<AuthFormFooter />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <AuthFormFooter link="/login" text="Already a Member?" />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
