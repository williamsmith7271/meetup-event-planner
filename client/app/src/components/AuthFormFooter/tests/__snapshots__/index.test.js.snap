exports[`<AuthFormFooter /> should render with default props 1`] = `
<Footer
  align="center"
  direction="row"
  justify="center"
  responsive={false}>
  <span>
    Already a Member?
      
  </span>
  <Anchor
    href="/login"
    tag="a">
    Login
  </Anchor>
</Footer>
`;
